### PR TITLE
Fixes missing context information in Range Index Config errors

### PR DESCRIPTION
**Pull request by adamretter**
Closes https://github.com/eXist-db/exist/issues/1339

### CODE REVIEW

File: extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfig.java [hunk 1]
```diff
@@ -96,8 +97,18 @@ private void parseChildren(NodeList configNodes, Map<String, String> namespaces)
...
-                } catch (DatabaseConfigurationException e) {
-                    LOG.error("Invalid range index configuration: " + e.getMessage());
+                } catch (final DatabaseConfigurationException e) {
+                    String uri = null;
+                    final Document doc = node.getOwnerDocument();
+                    if(doc != null) {
+                        uri = doc.getDocumentURI();
+                    }
+
+                    if(uri != null) {
```
Comment (dizzzz) [inline extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfig.java]: better style is `if (a==null) then {b} else {c}` in stead of `if (a!=null) then {c} else {b}` . But leave as is, as it is a minor-minor remark.

File: src/org/exist/dom/QName.java [hunk 2]
```diff
@@ -262,7 +262,7 @@ public static String extractLocalName(final String qname)
         } else if (p == qname.length()) {
             throw new IllegalArgumentException("Illegal QName: ends with a ':'");
         } else if (!isQName(qname)) {
-            throw new IllegalArgumentException("Illegal QName: not a valid local name.");
+            throw new IllegalArgumentException("Illegal QName: '" + qname + "' not a valid local name.");
```
Comment (dizzzz) [inline src/org/exist/dom/QName.java]: that will certainly help!

File: extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfig.java [hunk 3]
```diff
@@ -96,8 +97,18 @@ private void parseChildren(NodeList configNodes, Map<String, String> namespaces)
                     } else {
                         idxConf.add(newConfig);
                     }
-                } catch (DatabaseConfigurationException e) {
-                    LOG.error("Invalid range index configuration: " + e.getMessage());
+                } catch (final DatabaseConfigurationException e) {
+                    String uri = null;
+                    final Document doc = node.getOwnerDocument();
+                    if(doc != null) {
```
Comment (dizzzz) [inline extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfig.java]: I'd prefer a style like `String uri = (doc == null) ? null : doc.getDocumentURI();` but that is personal. dont change anything

**Comment by dizzzz** [comment]
ouch...

```
pile-test:
     [echo] Compiling tests 'index-range'
    [mkdir] Created dir: /home/travis/build/eXist-db/exist/extensions/indexes/range/build/test/classes
    [javac] Compiling 2 source files to /home/travis/build/eXist-db/exist/extensions/indexes/range/build/test/classes
    [javac] /home/travis/build/eXist-db/exist/extensions/indexes/range/test/src/org/exist/indexing/range/RangeIndexConfigTest.java:38: error: package org.exist.indexing.lucene does not exist
    [javac] import static org.exist.indexing.lucene.LuceneIndexConfig.MATCH_ATTR;
    [javac]                                        ^
    [javac] /home/travis/build/eXist-db/exist/extensions/indexes/range/test/src/org/exist/indexing/range/RangeIndexConfigTest.java:38: error: static import only from classes and interfaces
    [javac] import static org.exist.indexing.lucene.LuceneIndexConfig.MATCH_ATTR;
    [javac] ^
    [javac] /home/travis/build/eXist-db/exist/extensions/indexes/range/test/src/org/exist/indexing/range/RangeIndexConfigTest.java:39: error: package org.exist.indexing.lucene does not exist
    [javac] import static org.exist.indexing.lucene.LuceneIndexConfig.QNAME_ATTR;
    [javac]                                        ^
    [javac] /home/travis/build/eXist-db/exist/extensions/indexes/range/test/src/org/exist/indexing/range/RangeIndexConfigTest.java:39: error: static import only from classes and interfaces
    [javac] import static org.exist.indexing.lucene.LuceneIndexConfig.QNAME_ATTR;
    [javac] ^
    [javac] 4 errors
```

File: extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfig.java (hunk 1 above)
Comment (adamretter) [inline extensions/indexes/range/src/org/exist/indexing/range/RangeIndexConfig.java]: I tried that but it increased complexity and duplication

**Comment by adamretter** [comment]
@dizzzz Just fixed the Ant build, previously I was testing from IntelliJ

File: src/org/exist/dom/QName.java (hunk 2 above)
Comment (joewiz) [inline src/org/exist/dom/QName.java]: Yes - both showing the `qname` and surrounding it with quotes to make clear any extra whitespace.

**Comment by joewiz** [comment]
@adamretter Thank you! This will really help when debugging index configurations!

**Pull request merged**